### PR TITLE
Improve nydusify build subcommand

### PR DESCRIPTION
**Pull request by jiangliu**
Improve nydusify build subcommand and fix some resource leakages.

### CODE REVIEW

File: contrib/nydusify/pkg/backend/oss.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func newOSSBackend(rawConfig []byte) (*OSSBackend, error) {
 	accessKeySecret := configMap["access_key_secret"]
 	objectPrefix := configMap["object_prefix"]
 
-	if !ok1 || !ok2 {
-		return nil, fmt.Errorf("no endpoint or bucket is specified")
+	if !ok1 || endpoint == "" || !ok2 || bucketName == "" {
```
Comment (imeoer) [inline contrib/nydusify/pkg/backend/oss.go:52]: Maybe just check with `if endpoint == "" || bucketName == ""`.

File: contrib/nydusify/pkg/packer/pusher.go [hunk 2]
```diff
@@ -112,3 +112,13 @@ func ParseBackendConfig(backendConfigFile string) (BackendConfig, error) {
 	cfg.BlobPrefix = strings.TrimSuffix(cfg.BlobPrefix, "/")
 	return cfg, nil
 }
+
+func ParseBackendConfigString(backendConfigContent string) (BackendConfig, error) {
+	var cfg BackendConfig
+	if err := json.NewDecoder(strings.NewReader(backendConfigContent)).Decode(&cfg); err != nil {
```
Comment (imeoer) [inline contrib/nydusify/pkg/packer/pusher.go:118]: Maybe use `json.Unmarshal([]byte(backendConfigContent), &cfg)`.

File: contrib/nydusify/pkg/converter/cache.go [hunk 3]
```diff
@@ -149,11 +153,11 @@ func (cg *cacheGlue) PullBootstrap(
 	}
 
 	cacheRecord, bootstrapReader, blobReader, _ := cg.cache.Check(ctx, chainID)
+	defer bootstrapReader.Close()
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply contrib/nydusify/pkg/converter/cache.go:156]: Good catch, fixed

File: contrib/nydusify/cmd/nydusify.go [hunk 4]
```diff
@@ -441,99 +441,111 @@ func main() {
 			},
 		},
 		{
-			Name:  "pack",
-			Usage: "Pack a directory to nydus bootstrap and blob",
+			Name:    "build",
+			Aliases: []string{"pack"},
+			Usage:   "Build a Nydus image from a source directory",
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply contrib/nydusify/cmd/nydusify.go:446]: How abot "Build a Nydus filesystem from a source directory"?

File: contrib/nydusify/pkg/backend/oss.go (hunk 1 above)
Comment (jiangliu) [reply contrib/nydusify/pkg/backend/oss.go:52]: fixed

File: contrib/nydusify/pkg/packer/pusher.go (hunk 2 above)
Comment (jiangliu) [reply contrib/nydusify/pkg/packer/pusher.go:118]: fixed

File: contrib/nydusify/cmd/nydusify.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (imeoer) [reply contrib/nydusify/cmd/nydusify.go:446]: Looks good.

**Review by imeoer — APPROVED** [review]
Others LGTM, cc @hhhhsdxxxx PTAL.

**Comment by hhhhsdxxxx** [comment]
LGTM

**Pull request merged**